typescript/sdk/src/router/HyperlaneRouterChecker.ts

```diff
@@ -45,7 +45,7 @@ export class HyperlaneRouterChecker<
     const router = this.app.router(this.app.getContracts(chain));
 
     const checkMailboxClientProperty = async (
-      property: keyof MailboxClientConfig,
+      property: keyof Omit<MailboxClientConfig, 'ismFactoryAddresses'>,
```

Comment:
it looks like the zod schema change with `ismFactoryAddresses` makes it optional but this removes it entirely - is that your intention?

Reply:
Yup, there's a subtle reason for Omit. 

Its because we are using `keyof` which will create a type with all keys, despite it being optional. For this function, we do not want to include `ismFactoryAddresses` at all for `property`